kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerInstanceManager.java

```diff
@@ -375,10 +380,16 @@ private void remoteUpgradeContainer(final KieServicesClient client, final Contai
 
     RemoteKieServerOperation<Void> getContainersRemoteOperation(final ServerTemplate serverTemplate,
                                                                 final ContainerSpec containerSpec) {
+
         return new RemoteKieServerOperation<Void>() {
             @Override
             public Void doOperation(KieServicesClient client,
                                     Container container) {
+                
+                EnumSet<KieContainerStatus> set = EnumSet.of(KieContainerStatus.STARTED, KieContainerStatus.CREATING);
+                if(!set.contains(container.getStatus())) {
```

Comment:
What is the purpose of this check?

Reply:
if the container is in another state it does not make sense to retrieve the status (DISPOSE, STOPPED... all of them won't exists)